packages/headless/src/features/history/history-slice.ts

```diff
@@ -47,29 +47,64 @@ const isEqual = (current: HistoryState, next: HistoryState) => {
 const isContextEqual = (current: ContextState, next: ContextState) =>
   JSON.stringify(current.contextValues) === JSON.stringify(next.contextValues);
 
-// TODO: compare all facets non-idle values, comparing current values changes history after 1st request
-const isFacetsEqual = (current: FacetSetState, next: FacetSetState) =>
-  JSON.stringify(current) === JSON.stringify(next);
+type FacetStateWithCurrentValues = Record<
+  string,
+  CurrentValues<BaseFacetValueRequest>
+>;
 
-const isDateFacetsEqual = (
-  current: DateFacetSetState,
-  next: DateFacetSetState
-) => JSON.stringify(current) === JSON.stringify(next);
+const isFacetsEqual = (
+  current: FacetStateWithCurrentValues,
+  next: FacetStateWithCurrentValues
+) => {
+  for (const [key, value] of Object.entries(next)) {
+    if (!current[key]) {
+      return false;
+    }
 
-const isNumericFacetsEqual = (
-  current: NumericFacetSetState,
-  next: NumericFacetSetState
-) => JSON.stringify(current) === JSON.stringify(next);
+    const currentSelectedValues = current[key].currentValues.filter(
+      (value) => value.state === 'selected'
+    );
+    const nextSelectedValues = value.currentValues.filter(
+      (value) => value.state === 'selected'
+    );
+
+    if (
```

Comment:
Potential problem: If both next and current contain the same selected facet values, but in a different order, it should still be considered equal.

Because you would still get the same results from the index/same filter applied on the query.

I think the current check you have here (and for category facet) is dependant on the order.

Reply:
Yes exactly, I don't think the order of selected values should matter. If it does, modifying the method is trivial.